packages/core/admin/admin/src/core/apis/Components.js

```diff
@@ -0,0 +1,43 @@
+import { cloneDeep } from 'lodash';
+import invariant from 'invariant';
+
+// TODO this API should be merged with the components one
+class Components {
+  components = {};
+
+  getComponent = name => {
+    invariant(name, 'A name must be provided');
+
+    return cloneDeep(this.components[name]) || null;
```

Comment:
why do we want to clone deep instead of returning the reference in this case?

I would suggest to name the method `clone` if we want to stick with this implementation